README.md

```diff
@@ -1,2 +1,382 @@
 # github-actions
-Template repo that holds Github actions that can be reused by other repositories
+Template repo that holds Github actions that can be reused by other repositories.
+
+## Highlights
+
+- If a new repo needs to have actions set up for the first time, check out [Set Up Actions and Rules](#set-up-actions-and-rules).
+- To understand how a particular file works, check out the relevant link here:
+    - [Build .NET and Publish to NuGet](#how-build_dotnet_publish_nugetyml-works).
+- In everyday use:
+    - Open a PR to run checks for 1) semantic version increase compared to the most recent release, and 2) building in debug/release for both Ubuntu and Windows systems.
+    - Once all checks pass, merge the PR to build the project from the main branch and publish a pre-release package to the Github package manager.
+    - Create a new release that is tagged with the current version in main to publish a full package.
+        - This will build what is in the main branch, publish a full package to Github package manager, and also publish the full package to NuGet.
+        - Additionally, this will [announce the release on Discord](#link-github-to-discord-via-webhook).
+- Workflow calling and secrets.
+    - When another repo calls the workflows in this repo, it acts as if the workflow was copied and pasted into the other repo; that is, the "checkout" action will checkout the calling repo and not `github-actions`, even though this is the repo where the workflow exists. 
+    - To fully understand how secrets work with reusable workflows, check out [this link](https://docs.github.com/en/actions/sharing-automations/reusing-workflows#passing-inputs-and-secrets-to-a-reusable-workflow).
+
+## Set Up Actions and Rules
+
+- Copy the `example_caller_workflow.txt` and place it in the `.github/workflows` folder of the repository that needs to run the actions. As an example, this file will be placed in the `.github/workflows` folder for [bonsai-onix1](https://github.com/open-ephys/bonsai-onix1).
```

Comment:
why call it `example_caller_workflow.txt` and the require a rename instead of just calling it the yml file to begin with?

Reply:
I see, its so it does not run as action in this repo.